ambry-cloud/src/main/java/com.github.ambry.cloud/CloudDestination.java

```diff
@@ -37,6 +37,14 @@
   boolean uploadBlob(BlobId blobId, long inputLength, CloudBlobMetadata cloudBlobMetadata, InputStream blobInputStream)
       throws CloudStorageException;
 
+  /**
+   * Download blob from the cloud destination.
+   * @param blobId id of the Ambry blob to be downloaded
+   * @param outputStream outputstream to populate the downloaded data with
+   * @throws CloudStorageException if the download encounters an error.
+   */
+  void downloadBlob(BlobId blobId, OutputStream outputStream) throws CloudStorageException;
```

Comment:
Was expecting this to return CloudBlobMetadata.  Does that require a separate call?

Reply:
This doesn't return a CloudBlobMetadata because we already obtained it during BlobStore::get. This is the actual download of blob from blob store, which happens either during prefetch or when the response is being returned to frontend.

Reply:
What you're saying makes sense.  But I'd rather not see the interface contract be designed based on the particulars of how the callers are using it.  This is okay for the recovery flow.  But when we are live serving blobs from Azure, a GET request will want to make a single call to get blob data and metadata (no initial call to getBlobMetadata().

Reply:
Sure.. I will keep that in mind. Also I will try to see if we can revisit this logic for the recovery too in future iteration.